
syntax = "proto3";

package proto;

import "map_limits.proto";

message Grid2D {
	message CellBox {
		int32 max_x = 1;
		int32 max_y = 2;
		int32 min_x = 3;
		int32 min_y = 4;
	}

	MapLimits limits = 1;
  // These values are actually int16s, but protos don't have a native int16
  // type.
	repeated int32 cells = 2;
	CellBox konwn_cell_box = 3;
	float min_correspondence_cost = 4;
	float max_correspondence_cost = 5;
}

message HybridGrid {
  float resolution = 1;
  // '{x, y, z}_indices[i]' is the index of 'values[i]'.
  repeated sint32 x_indices = 3;
  repeated sint32 y_indices = 4;
  repeated sint32 z_indices = 5;
  // The entries in 'values' should be uint16s, not int32s, but protos don't
  // have a uint16 type.
  repeated int32 values = 6;	
}